tests/client/test_client.py

```diff
@@ -19,6 +19,24 @@ def test_get(server):
     assert repr(response) == "<Response [200 OK]>"
 
 
+def test_build_request(server):
+    url = server.url.copy_with(path="/echo_headers")
+    headers = {"Custom-header": "value"}
+
+    with httpx.Client() as http:
+        request = http.build_request("GET", url)
+        request.headers.update(headers)
+        response = http.send(request)
+
+    assert response.status_code == 200
+    assert response.url == url
+
+    response_data = response.json()
```

Comment:
Since we're only asserting one value we can simplify our asserts like this: `assert response.json()["custom-header"] == "value"`